src/clj/game/cards/events.clj

```diff
@@ -522,6 +522,32 @@
     :msg "look at the top 4 cards of their Stack and add 1 of them to their Grip"
     :effect (effect (move target :hand) (shuffle! :deck))}
 
+   "Falsified Credentials"
+   {:prompt "Choose a type"
+    :choices ["Agenda" "Asset" "Upgrade"]
+    :effect (effect
+             (resolve-ability
+              (let [chosen-type target]
+                {:choices {:req #(let [topmost (get-nested-host %)]
+                                   (and (is-remote? (second (:zone topmost)))
+                                        (= (last (:zone topmost)) :content)
+                                        (not (:rezzed %))))}
+                 :delayed-completion true
+                 :effect (req             ;taken from Drive By - maybe refactor
+                          (when-completed (expose state side target) 
+                            (if async-result ;; expose was successful
+                              (if (= chosen-type (:type target))
+                                (do (resolve-ability
```

Comment:
This `resolve-ability` and following `effect-completed` can be merged into a `continue-ability` that no longer needs the `do` also!